### PR TITLE
[framework] show information about project in debug toolbar

### PR DESCRIPTION
**Pull request by henzigo**
| Q             | A
| ------------- | ---
|Description, reason for the PR| After I started work with more projects, I was asking my colleagues to tell me which version of framework project has. So I created toolbar with version and basic information about project. I also upgraded releaser which will update version automatically.
|New feature| Yes 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

Screenshot 2019-05-11 at 23 50 44

### CODE REVIEW

**Comment by PetrHeinz** [comment]
Nice idea! :+1: 

There are few things that need improvement before merging imo:

1) The logo in debug bar should be white, ideally in svg instead of png.
2) I would use the name "Shopsys Framework Knowledge Base" or "SSFW Knowledge Base" instead of "Framework Documentation".
3) Not sure about the "dev-master" version. There is no guarantee that it's dev-master, just that it's some unreleased version... Maybe it should be called "7.2.0-dev" (when the last released version was 7.1.0), similarly as in the [Symfony Kernel class](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/Kernel.php). 
4) The release worker should not only change it to the released version during release-candidate phase, but also back to "dev-master" (or to "<next-version-number>-dev" as suggested above).
5) The classes `VersionUpgradeFileManipulator` and `UpdateUpgradeReleaseWorker` seem to be responsible for updating the upgrade notes. I think that this functionality should be separated into its own worker.
6) The `ShopsysCollector` (maybe rather `ShopsysFrameworkDataCollector`?) should fill the `data` property in the `collect` method, not in constructor. See eg. `RedisDataCollector`, `ConfigDataCollector` or `EventDataCollector`.
7) The `reset` method should set the `data` property to an empty array.

**Comment by henzigo** [comment]
Hello, here is my answers to your comment:

1.  Logo is white now
2. "Shopsys Framework Knowledge Base" is too long for me and maybe "documentation" word is better to understand than "knowledge base". Secondly, we don't use SSFW often. I'm not changing it for now, but you can change it if you want.
3. I did it by you, so now its version 7.2.0-dev
4. Dev version will be now set after release (7.2.0 -> 7.3.0-dev)
5. I created new workers and file manipulator
6. updated
7. updated

I would like to see that function in next version, so please look at it and try to merge it :) @PetrHeinz

**Comment by henzigo** [comment]
It looks awesome now 👍 
Screenshot 2019-05-13 at 23 21 31

**Comment by henzigo** [comment]
Guys, it would be nice to have it in next minor release. What if programmer will have to write next version number on releaser request. However, the programmer can always adjust release version manually.

File: packages/framework/src/Resources/views/Components/Collector/shopsysFramework.html.twig [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                <b>Locale</b>
+                <span class="sf-toolbar-status">{{ collector.currentDomainLocale }}</span>
+            </div>
+        </div>
+        <div class="sf-toolbar-info-group">
+            {% for domain in collector.domains %}
+                <div class="sf-toolbar-info-piece">
+                    <b>{{ domain.name }}</b>
+                    <span>
+                        {% if domain.id == collector.currentDomainId %}
```
Comment (TomasLudvik) [inline packages/framework/src/Resources/views/Components/Collector/shopsysFramework.html.twig]:
```suggestion
                        {% if domain.id is same as(collector.currentDomainId) %}
```

File: packages/framework/src/Resources/views/Components/Collector/shopsysFramework.html.twig [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                            <a href="{{ domain.url }}">{{ domain.url }}</a>
+                        {% endif %}
+                    </span>
+                </div>
+            {% endfor %}
+        </div>
+        <div class="sf-toolbar-info-group">
+            <div class="sf-toolbar-info-piece">
+                <b>Resources</b>
+                <span><a href="https://github.com/shopsys/shopsys/blob/master/docs/index.md" target="_blank" rel="help">Framework Documentation</a></span>
```
Comment (TomasLudvik) [inline packages/framework/src/Resources/views/Components/Collector/shopsysFramework.html.twig]:
```suggestion
                <span><a href="https://github.com/shopsys/shopsys/blob/master/docs/index.md" target="_blank" rel="help">Documentation</a></span>
```

File: utils/releaser/src/FileManipulator/FrameworkVersionFileManipulator.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+{
+    /**
+     * @var string
+     */
+    public const FRAMEWORK_VERSION_FILE_PATH = '/packages/framework/src/ShopsysFrameworkBundle.php';
+
+    /**
+     * @var string
+     */
+    private const FRAMEWORK_VERSION_PATTEN = "/public const VERSION = '(.+)';/";
```
Comment (TomasLudvik) [inline utils/releaser/src/FileManipulator/FrameworkVersionFileManipulator.php]:
```suggestion
    private const FRAMEWORK_VERSION_PATTERN = "/public const VERSION = '(.+)';/";
```

File: packages/framework/src/ShopsysFrameworkBundle.php [hunk 4]
```diff
@@ -13,6 +13,11 @@
 
 class ShopsysFrameworkBundle extends Bundle
 {
+    /**
+     * @var string
+     */
+    public const VERSION = '7.2.0-dev';
```
Comment (TomasLudvik) [inline packages/framework/src/ShopsysFrameworkBundle.php]: The really cool would be to have link to documentation here. 

UpdateFrameworkKernelVersionReleaseWorker would update it to something like `https://github.com/shopsys/shopsys/blob/v7.2.0/docs/index.md` and `SetDevFrameworkKernelVersionReleaseWorker` back to `https://github.com/shopsys/shopsys/blob/master/docs/index.md`

Comment (grossmannmartin) [reply packages/framework/src/ShopsysFrameworkBundle.php]: Hi, I solved it differently to not need to update another thing in release worker and it's bounded to the version constant now.

File: packages/framework/src/Resources/views/Components/Collector/shopsysFramework.html.twig (hunk 2 above)
Comment (grossmannmartin) [reply packages/framework/src/Resources/views/Components/Collector/shopsysFramework.html.twig]: Applied manually as I updated the file in the meantime

File: packages/framework/src/ShopsysFrameworkBundle.php (hunk 4 above)
Comment (TomasLudvik) [reply packages/framework/src/ShopsysFrameworkBundle.php]: Your solution is much better than mine, good job!

**Comment by simara-svatopluk** [comment]
The UI looks great!

I'm unfortunatelly unable to test the release process becasue step 9 `php phing translations-dump` fails.
Please fix it first (maybe rebase to current working branch may help).

**Comment by PetrHeinz** [comment]
@simara-svatopluk Please, fix the typo in the annotation directly in vendor yourself. Currently we can either use the fix in #1169 or wait for new release of `studio-42/elfinder`.

**Comment by simara-svatopluk** [comment]
I tried to release version `8.0.0-alpha1`. The RC worker works fine, turns the version to `8.0.0-alpha1`.

But the after release worker turns the version to `8.1.0-dev`, but that's wrong in this case. In this case I would expected `8.0.0-alpha2` or `8.0.0-beta` , `8.0.0` or `8.0.0-dev`, but none of them automatically... I expect to have a possibility to choose desired version.

So this "after release" behavior is wrong in my opinion and I suggest to change the worker to just a message `[Manually] Change the version in ShopsysFrameworkBundle::VERSION, suggesting "8.0.0-dev", "8.1.0-dev" and commit the change`

**Comment by PetrHeinz** [comment]
@simara-svatopluk I think that after the release of `8.0.0-alpha1` you should be reverted to `8.0.0-dev`. But, considering the releaser is not prepared for working on different dev branches, I understand where the assumption that a new minor release is being prepared came from.

However, I like your idea of a manual release worker to avoid the problem altogether. Maybe with a suggestion of a new `MAJOR`, `MINOR` and `PATCH` dev version (or if the `PreReleaseSuffix` is present, suggest the dev version with same version number).

---

Seeing the output and commit messages I'd like to ask whether we really want to call it "shopsys framework kernel version" and if so, whether we shouldn't do it more consistently (see "shopsys kernel" / "framework kernel"). I'd rather just call it "SSFW version" or "Shopsys Framework version".

**Comment by grossmannmartin** [comment]
@PetrHeinz hmm... You're right about inconsistency. 
the thing that bugs me most is a commit message (release worker make a commit)
Right now it's like `shopsys kernel version is now updated to version "8.0.0-dev"`, but changing it to Shopsys Framework change the meaning... (`shopsys framework is now updated....`)

**Comment by PetrHeinz** [comment]
@grossmannmartin How about `ShopsysFrameworkBundle: version updated to "8.0.0-dev"`?

**Comment by grossmannmartin** [comment]
@PetrHeinz Like.
I updated the names accordingly and should be unified (didn't touch the commits as they will be squashed)

File: utils/releaser/src/ReleaseWorker/AfterRelease/SetFrameworkBundleVersionToDevReleaseWorker.php [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            if (!$version->hasPreReleaseSuffix()) {
+                throw new RuntimeException(
+                    'Development version should be suffixed (with \'-dev\', \'-alpha1\', ...)'
+                );
+            }
+
+            return $version;
+        });
+
+        $developmentVersion = $this->symfonyStyle->askQuestion($question);
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply utils/releaser/src/ReleaseWorker/AfterRelease/SetFrameworkBundleVersionToDevReleaseWorker.php]: Applied manually as more work had to be done.

**Pull request merged**